test/unit/index.js

```diff
@@ -48,6 +41,55 @@ describe("asyncPool", function() {
         await asyncPool(2, [100, 500, 300, 200], timeout);
         expect(results).to.deep.equal([100, 300, 500, 200]);
       });
+
+      it("runs all promises in parallel when the pool is bigger than needed", async function() {
+        const results = [];
+        const timeout = i =>
+          new Promise(resolve =>
+            setTimeout(() => {
+              results.push(i);
+              resolve();
+            }, i)
+          );
+        await asyncPool(5, [100, 500, 300, 200], timeout);
+        expect(results).to.deep.equal([100, 200, 300, 500]);
+      });
+
+      it("rejects on error with pool limit effective", async function() {
+        const results = [];
+        const timeout = i =>
+          new Promise((resolve, reject) =>
+            setTimeout(() => {
+              results.push(i);
+
+              if (i === 300) {
+                reject(new Error("Oops"));
+              } else {
+                resolve();
+              }
+            }, i)
+          );
+        await rejects(asyncPool(2, [100, 500, 300, 200], timeout));
+      });
+
+      it("rejects on error when pool bigger than needed", async function() {
+        const results = [];
+        const timeout = i =>
+          new Promise((resolve, reject) =>
+            setTimeout(() => {
+              results.push(i);
+
+              if (i === 300) {
+                reject(new Error("Oops"));
+              } else {
+                resolve();
+              }
+            }, i)
```

Comment:
Is this test still equivalent in case we do the following simplification?
```diff
  setTimeout(() => {
-   results.push(i);
- 
-   if (i === 300) {
-     reject(new Error("Oops"));
-   } else {
-     resolve();
-   }
- }, i)
+   reject(new Error("Oops"));
+ })
```

Reply:
Well, my intention was to have a setup where only one of the promises fails, to show that this is sufficient to make the entire call being rejected. But actually it does not make a big difference with the code under test. 
How about simplifying it even further:
`const timeout = _ => Promise.reject();`
?

Reply:
:+1: for simplifying it even further